ql/ql/src/queries/style/MissingParameterInQlDoc.ql

```diff
@@ -0,0 +1,85 @@
+/**
+ * @name Missing QLDoc for parameter
+ * @description It is suspicious when a predicate has a parameter that is
+ *              unmentioned in the qldoc, and the qldoc contains a
+ *              code-fragment mentioning a non-parameter.
+ * @kind problem
+ * @problem.severity warning
+ * @precision high
+ * @id ql/missing-parameter-qldoc
+ * @tags maintainability
+ */
+
+import ql
+
+/**
+ * Gets a fragment enclosed in backticks (`) from a QLDoc of the predicate `p`.
+ * Skips code-blocks that are triple-quoted.
+ */
+private string getACodeFragment(Predicate p) {
+  result = p.getQLDoc().getContents().regexpFind("`([^`]*)`(?!`)", _, _)
+}
+
+/** Gets a parameter name from `p`. */
+private string getAParameterName(Predicate p) { result = p.getParameter(_).getName() }
+
+/**
+ * Gets the name of a parameter of `p` that is mentioned in any way in the QLDoc of `p`.
+ * Also includes names that are mentioned in non-code fragments.
+ */
+private string getADocumentedParameter(Predicate p) {
+  result = p.getQLDoc().getContents().regexpFind("\\b\\w[\\w_]*\\b", _, _) and
+  result.toLowerCase() = getAParameterName(p).toLowerCase()
+}
+
+/**
+ * Get something that looks like a parameter name from the QLDoc,
+ * but which is not a parameter of `p`.
+ */
+private string getAMentionedNonParameter(Predicate p) {
+  exists(string fragment | fragment = getACodeFragment(p) |
+    result = fragment.substring(1, fragment.length() - 1)
+  ) and
+  result.regexpMatch("^[a-z]\\w+$") and
+  not result.toLowerCase() = getAParameterName(p).toLowerCase() and
+  not result = ["true", "false", "NaN", "this"] and // keywords
```

Comment:
I would add `forall` and `exists`, and possibly the names of various aggregates to this mix.